fc/mef.py

```diff
@@ -217,265 +52,152 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
     Notes
     -----
     GMM finds clusters by fitting a linear combination of `n_clusters`
-    Gaussian probability density functions (pdf) to `data`, by likelihood
-    maximization.
+    Gaussian probability density functions (pdf) to `data` using
+    Expectation Maximization (EM).
+
+    GMM can be fairly sensitive to the initial parameter choice. To
+    generate a reasonable set of initial conditions, `clustering_gmm`
+    first divides all samples in `data` into `n_clusters` groups of the
+    same size based on their Euclidean distance to the origin. Then, for
+    each group, the 50% samples farther away from the mean are discarded.
+    The mean and covariance are calculated from the remaining samples of
+    each group, and used as initial conditions for the GMM EM algorithm.
 
     `clustering_gmm` internally uses `GMM` from the ``scikit-learn``
-    library. For more information, consult their documentation.
+    library, with full covariance matrices for each cluster and a fixed,
+    uniform set of weights. This means that `clustering_gmm` implicitly
+    assumes that all bead subpopulations have roughly the same number of
+    events. For more information, consult ``scikit-learn``'s documentation.
```

Comment:
Throwing away samples 50% away from cluster centers seems pretty arbitrary and unjustified. I assume this is all part of the `scikit-learn` GMM bug? In which case this is kinda a hack to get around that? I've already raised my concerns regarding that bug in #159, and it may not be worth fixing at this point in time. In general, though, it would be nice to get rid of this strange initialization procedure if the `scikit-learn` GMM bug was ever fixed. In my previous experience (MATLAB implementation), a vanilla GMM fit would work fine with a reasonable K-means initialization.

I would also prefer not to hardcode the weights equally. If the GMM algorithm works as advertised, that should be a pretty straightforward parameter to fit, and leaving it unbounded allows for edge cases where you have non-uniform subpopulation sizes (again, the devil's advocate argument that immediately comes to mind is if you mixed two different bead samples and were unable to guarantee equal ratio between the two; just feels like an unnecessary assumption). I feel like this also relates to the GMM bug, though, in which case it may not be worth fixing now and would warrant a deeper investigation into the inner workings of `scikit-learn`'s GMM implementation.

If this works good enough now, then I probably wouldn't change these things.